app/models/role.js

```diff
@@ -1,6 +1,17 @@
+import Ember from 'ember';
 import Model from 'ember-data/model';
 import attr from 'ember-data/attr';
+import { hasMany } from 'ember-data/relationships';
 
 export default Model.extend({
-  name: attr('string')
+  name: attr('string'),
+  rolePermissions: hasMany('rolePermission', { async: false }),
+
+  permissions: Ember.computed('rolePermissions.[]', function(){
+    return this.get('rolePermissions').map(rp => rp.get('permission'));
```

Comment:
Ember provides `getEach` options for Array. Would be better to use it here instead of map?